src/main/java/io/cryostat/rules/RuleProcessor.java

```diff
@@ -238,6 +238,7 @@ private void startRuleRecording(ConnectionDescriptor connectionDescriptor, Rule
                             recordingTargetHelper.parseEventSpecifierToTemplate(
                                     rule.getEventSpecifier());
                     recordingTargetHelper.startRecording(
+                            true,
```

Comment:
I forgot the explanation from yesterday. Is this set to `true` so that every time a `Rule` is processed, if it has already been previously processed and a recording already exists, instead of throwing an `Exception` the old recording is 'quietly' replaced with a new one that is exactly the same?

Reply:
Yea exactly. This is only done for recordings (re)started by rules, not ones created explicitly and manually by a user. Without this, if a user created rule `A` which in turn caused the creation and start of rule R, and then the user deleted `A` with cleanup, then `R` would be stopped and remain in the target. If the user then tries to (re)create a rule named `A'` that has the same definition as the original `A`, the rule would never be able to start a new recording `R'` because `R` is still there in a stopped state.

Reply:
Okay, that makes sense. Is there a reason for stopping the `Rule` recording instead of simply closing (i.e. deleting) it?

Reply:
Actually, I guess this way the user can still create a snapshot or dump the recording to archives.

Reply:
It's more or less harmless to leave the recording in place but stopped - it'll remove any additional resource overhead that JFR was consuming while the recording ran, and the in-memory JFR ring buffers are pre-allocated regardless, so the only additional expense involved in stopping rather than deleting is that some disk space may be taken up if the recording was configured to also transfer chunks to disk. Since the cost is low and configurable, I figured it was best to leave the data there in case the user does still want it in the future, rather than delete it and have it lost forever if they haven't performed an archive on it in advance.